R/boxr_auth.R

```diff
@@ -629,3 +636,71 @@ get_token <- function() {
   
   stop("No token available", call. = FALSE)
 }
+
+#' Harmonize token location
+#' 
+#' The canonical location for tokens is in the `~` directory. Unfortunately,
+#' `~` resolves differently on Windows for R than it does for other languages.
+#' This has been a source of friction. The solution implemented by the r-lib
+#' team is that `fs::path_expand()` resolves only to the home directory.
+#' `fs::path_expand_r()` behaves like `path.expand()`, as `~` resolves to the 
+#' `Documents` folder, rather than the home folder.
+#' 
+#' boxr was built using the `path.expand()` philosophy; we are moving to the 
+#' r-lib philosophy.
+#' 
+#' The purpose of this function is to help manage the transition.
+#' 
+#' This function will look for the cache file in both locations. If the file is 
+#' found in the "wrong" location, it will offer to move it to the "right" 
+#' location.
+#' 
+#' It returns the absolute path to the cache file, if it exists.
+#'  
+#' @param cache `character` path to cache, unexpanded
+#' 
+#' @return `character` absolute path to cache
+#' 
+#' @noRd
+#'
+harmonize_token_location <- function(cache) {
+
+  path_cache <- fs::path_expand(cache)
+  path_r_cache <- fs::path_expand_r(cache)
+  
+  if (fs::file_exists(path_cache)) {
+    # success
+    return(path_cache)
+  }
+  
+  if (fs::file_exists(path_r_cache)) {
+    
+    # file in "bad" location, offer to move it
+    if (interactive()) {
+      msg <- glue::glue(
+        "Token file found at `{path_r_cache}`.",
+        "Move token file to new boxr standard location `{path_cache}`? (Y/n): ",
+        sep = "\n"
+      )
+      str_move <- readline(prompt = msg)
+      
+      # we are moving the file
+      if (substr(str_move, 1, 1) %in% c("Y", "y", "")) {
+        fs::file_move(path_r_cache, path_cache)
+        path <- path_cache
```

Comment:
is `path` used later?

Reply:
dooh! I had refactored things, but did not clean up after myself, thanks!